### PR TITLE
chore: update dependencies

### PR DESCRIPTION
**Pull request by avivkeller**
## Description

This PR updates all dependencies to the latest versions they support.

Note that ESLint is intentionally kept at `9.14.0` instead of `9.15.0`, see https://github.com/nodejs/nodejs.org/pull/7232#issuecomment-2480209160

### Check List

- [x] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [x] I have run `npm run format` to ensure the code follows the style guide.
- [x] I have run `npm run test` to check if all tests are passing.
- [x] I have run `npx turbo build` to check if the website builds without errors.

### CODE REVIEW

**Comment by bmuenzenmeyer** [comment]
@RedYetiDev in case you were unaware, there is some tenuous intentionality to our dependency pinning

https://github.com/nodejs/nodejs.org/blob/main/DEPENDENCY_PINNING.md

I have always been in the camp that we should pin ALL of them, but you don't win em all 😉

**Comment by avivkeller** [comment]
I was aware. According to that document, "^" and "~" are both used, but I'll revert that change (if something changes, there can always be a follow up)

**Comment by ovflowd** [comment]
> I was aware. According to that document, "^" and "~" are both used, but I'll revert that change (if something changes, there can always be a follow up)

Yup please revert the ^ to ~

File: apps/site/package.json [hunk 1]
```diff
@@ -47,52 +47,52 @@
...
-    "@vercel/speed-insights": "~1.1.0",
-    "autoprefixer": "~10.4.20",
-    "classnames": "~2.5.1",
+    "@vcarl/remark-headings": "^0.1.0",
+    "@vercel/analytics": "^1.4.0",
+    "@vercel/otel": "^1.10.0",
+    "@vercel/speed-insights": "^1.1.0",
+    "autoprefixer": "^10.4.20",
+    "classnames": "^2.5.1",
     "critters": "0.0.23",
```
Comment (ovflowd) [inline apps/site/package.json]: This dependency needs to be removed, not used anymore.

**Comment by avivkeller** [comment]
I'm working on updating the packages, but I'm having some errors, so I'm figuring out which dep to not update:
```
apps/site/components/Common/Search/States/WithSearchBox.tsx:56:13 - error TS2345: Argument of type 'Dispatch<SetStateAction<SearchResults>>' is not assignable to parameter of type '(value: Nullable<Results<DocumentType>>) => void | PromiseLike<void>'.
  Types of parameters 'value' and 'value' are incompatible.
    Type 'Nullable<Results<DocumentType>>' is not assignable to type 'SetStateAction<SearchResults>'.
      Type 'Results<DocumentType>' is not assignable to type 'SetStateAction<SearchResults>'.
        Type 'Results<DocumentType>' is not assignable to type 'Results<SearchDoc>'.
          Type 'DocumentType' is missing the following properties from type 'SearchDoc': id, path, pageTitle, siteSection, and 2 more.

56       .then(setSearchResults)
               ~~~~~~~~~~~~~~~~

apps/site/components/Common/Search/States/WithSearchBox.tsx:77:29 - error TS2345: Argument of type 'Dispatch<SetStateAction<SearchResults>>' is not assignable to parameter of type '(value: Nullable<Results<DocumentType>>) => void | PromiseLike<void>'.
  Types of parameters 'value' and 'value' are incompatible.
    Type 'Nullable<Results<DocumentType>>' is not assignable to type 'SetStateAction<SearchResults>'.
      Type 'Results<DocumentType>' is not assignable to type 'SetStateAction<SearchResults>'.
        Type 'Results<DocumentType>' is not assignable to type 'Results<SearchDoc>'.
          Type 'DocumentType' is missing the following properties from type 'SearchDoc': id, path, pageTitle, siteSection, and 2 more.

77     getInitialFacets().then(setSearchResults).catch(setSearchError);
                               ~~~~~~~~~~~~~~~~

apps/site/components/MDX/SearchPage/index.tsx:59:26 - error TS2345: Argument of type 'Nullable<Results<DocumentType>>' is not assignable to parameter of type 'SetStateAction<SearchResults>'.
  Type 'Results<DocumentType>' is not assignable to type 'SetStateAction<SearchResults>'.
    Type 'Results<DocumentType>' is not assignable to type 'Results<SearchDoc>'.
      Type 'DocumentType' is missing the following properties from type 'SearchDoc': id, path, pageTitle, siteSection, and 2 more.

59         setSearchResults(results);
                            ~~~~~~~

apps/site/components/MDX/SearchPage/index.tsx:60:36 - error TS2345: Argument of type '(Hit | Result<DocumentType>)[]' is not assignable to parameter of type 'Hit[]'.
  Type 'Hit | Result<DocumentType>' is not assignable to type 'Hit'.
    Type 'Result<DocumentType>' is not assignable to type 'Hit'.
      Types of property 'document' are incompatible.
        Type 'DocumentType' is missing the following properties from type 'SearchDoc': id, path, pageTitle, siteSection, and 2 more.

60         setHits(hits => uniqueHits([...hits, ...(results?.hits ?? [])]));
                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 4 errors.

husky - pre-commit script failed (code 1)
```

**Comment by avivkeller** [comment]
It was Orama. I'll do a follow-up figuring out other things that need to be changed to facilitate that update.

**Comment by araujogui** [comment]
I am seeing `Unicode reference out of bounds` errors both locally and CI 🤔 I'm looking into it

**Comment by araujogui** [comment]
It's `shiki`, rolling it back on the build works.

**Review by araujogui — APPROVED** [review]

**Review by AugustinMauroy — APPROVED** [review]

**Pull request merged**